cg-spec/editors_draft.html

```diff
@@ -2671,12 +2671,36 @@ <h2>IANA Considerations</h2>
 
-    <p>In the meantime, implementers of RDF-star are faced with a dilemma: use the standard media-type identifiers as if their definition had been already updated (optimistic approach); or make it explicit that they are prone to produce or consume embedded triples (pessimistic approach), perhaps by using a different temporary media-type from the unregistered `x.` tree [[RFC6838]], profile links [[RFC6906]], or custom HTTP headers.</p>
+    <p>In the meantime, implementers of RDF-star are faced with a dilemma: use the standard media-type identifiers as if their definition had been already updated (optimistic approach); or make it explicit that they are prone to produce or consume embedded triples (pessimistic approach), perhaps by using a different temporary media-type from the unregistered `x-` tree [[RFC6838]], profile links [[RFC6906]], or custom HTTP headers.</p>
 
     <p>The optimistic approach raises the risk of breaking the expectations of other applications that are not RDF-star-aware. For example, if a SPARQL-star endpoint responds to a CONSTRUCT query with the `text/turtle` media-type, and includes an <a>embedded triple</a> in the response, a legacy client will fail to parse that response. On the other hand, if the same server applies the pessimistic approach, it may simply reject any query requiring `text/turtle`, <em>just in case</em> the result contain <a>embedded triples</a>. Note that it is not always feasible for the server to decide beforehand whether the result is plain Turtle or Turtle-star, because the result is often produced in a streamed manner, after the headers containing the media-type have been sent to the client.</p>
 
     <p>Another problem with the pessimistic approach is that "temporary" media-types are known to actually become permanent. Old applications keep using them long after their intended expiration date. This a burden on all other implementations, which must handle the "temporary" media-types along the permanent one, as aliases, in order to maintain backward compatibility.</p>
 
+    <p>The W3C Recommendation 
+      <a data-cite="DX-PROF-CONNEG#introduction">Content Negotiation by Profile</a>
+      [[DX-PROF-CONNEG]] describes a mechanism that separates content-type
+      negotiation from requesting a response conforming to a specific information model,
+      whether that model is a specific standard, specification or profile.
+      The <a data-cite="DX-PROF-CONNEG#requestsandresonses"
+             >operations of the abstract model</a> 
+      are "list profiles" and "get resource by profile".
+      The "get resource" operation can provide the profile requests with HTTP
+      header <code>Accept-Profile</code> and responses use <code>Content-Profile</code>, or
+      alternatively the request can use HTTP Query String parameters.
```

Comment:
How does this address the clear statement (I don't have the citation handy; sorry) that profiled media types are supposed to be subsets of the media type, i.e., `<< >>` notation is not valid `text/turtle` regardless of the profile attached -- _because `<< >>` notation **is not valid** `text/turtle`!_

Reply:
@TallTed  I agree. However, the way I see it, a future working group could
- extend text/turtle to support turtle-star 
- define different profiles for the text/turtle (with or without RDF-star, possibly others)
- define that the default profile SHOULD be "without RDF-star". Since old client will not require any profile, this would encourage servers to accommodate them if practical.

As Andy wrote, this does not solve the optimistic/pessimistic dilemma for us. But this should be considered by a future WG in the design of a durable solution.

Reply:
The negotiation is of information model and "profile" in that document is a "specification that constrains, extends, combines, or provides guidance or explanation about the usage of other specifications."

That said, TR/dx-prof-conneg is of possible if `text/turtle` is updated. Media types are not immutable and other types have changed "in place" over the years or bypassed the media type issues and done their own thing (e.g. XML 1.1).

As we have identified, there is no perfect solution here. Each route has negative impact in some cases. Until we have field experience, we can't make a fact-based decision.

Reply:
> That said, `TR/dx-prof-conneg` is [only] possible if `text/turtle` is updated. Media types are not immutable and other types have changed "in place" over the years or bypassed the media type issues and done their own thing (e.g. XML 1.1).
> 
> As we have identified, there is no perfect solution here. Each route has negative impact in some cases. Until we have field experience, we can't make a fact-based decision.

Fair enough. But since we can't make a fact-based decision, I submit that we should not make a decision at all, but rather list the possible options we know of, along with the pros and cons and additional requirements of each — such that it's clear that, for instance, `TR/dx-prof-conneg` could be brought into play (and solve our media type quandary) *IFF* `text/turtle` (and some other current media types) were also updated to have a profile which resulted in today's format and which was the effective profile when no profile was declared, *along with* such other profiles as were deemed needed/relevant at that point, such as (hopefully) profiles for RDF-star serializations, such as (the most relevant for this example) Turtle-star.

Reply:
I don't believe it makes a decision.

The section is "non-normative".  

The new text says "This may be useful when the information is available in RDF-star or in non-RDF-star RDF 1.1 form," and refers back to the dilemma set out in the beginning of the section.

Reply:
A section being non-normative does not mean it will not be take as a strong suggestion, which can amount to a decision even though authors and editors and others loudly assert that it was not a decision.

I think that links back to these discussions/issues/PRs would be as appropriate, possibly more so, as the links to RFCs we're not deciding to use but suggesting as possibilities...

While minimizing the possibility of confusion, or suggestion of debate, or even word count, can sometimes be appropriate in WG Candidate or Proposed Recommendations, I do not think the same applies to CG Reports, especially where we (at least, I) anticipate significant discussion and conclusions that may or may not include any of the possibilities listed in the report, due to the anticipated (or at least hoped for) broader community participation.